### PR TITLE
ipc4: adjust max ipc size for ipc4

### PR DESCRIPTION
**Pull request by RanderWang**
The max ipc size of 384 is a legacy issue. We first wanted
to send ipc message in multiple-thread style, but later
we have to serialize ipc message and we didn't change max
ipc size to fully utilize memory window. Now for ipc4,
we have to set it to min(dspbox, hostbox) to align with
host driver.

Signed-off-by: Rander Wang <rander.wang@intel.com>

### CODE REVIEW

File: src/include/sof/ipc/topology.h [hunk 1]
```diff
@@ -39,13 +39,20 @@ typedef uint32_t ipc_comp;
...
+
 #elif CONFIG_IPC_MAJOR_4
 #include <ipc4/pipeline.h>
 #include <ipc4/module.h>
 #include <ipc4/gateway.h>
 #define ipc_from_pipe_new(x) ((struct ipc4_pipeline_create *)x)
 #define ipc_from_pipe_connect(x) ((struct ipc4_module_bind_unbind *)x)
 
+/* Maximum message size for mailbox Tx/Rx based on mailbox size */
+#define SOF_IPC_MSG_MAX_SIZE                  0x1000
```
Comment (lgirdwood) [inline src/include/sof/ipc/topology.h:54]: Can we explain which mailbox part (macro name) we use for max size

File: src/include/sof/ipc/topology.h [hunk 2]
```diff
@@ -39,13 +39,20 @@ typedef uint32_t ipc_comp;
 #define ipc_from_pipe_connect(x) ((struct sof_ipc_pipe_comp_connect *)x)
 #define ipc_from_comp_new(x) ((struct sof_ipc_comp *)x)
 #define ipc_from_dai_config(x) ((struct sof_ipc_dai_config *)x)
+
+/* Maximum message size for mailbox Tx/Rx */
+#define SOF_IPC_MSG_MAX_SIZE                    384
```
Comment (lgirdwood) [inline src/include/sof/ipc/topology.h:44]: No reason why this can be increased as a separate patch.

File: src/include/sof/ipc/topology.h (hunk 1 above)
Comment (RanderWang) [reply src/include/sof/ipc/topology.h:54]: yes, this is a little tricky. we have dspbox and hostbox with different size (0x1000, 0x2000), but we check ipc message size with  SOF_IPC_MSG_MAX_SIZE.  We should unify the size of dspbox & hostbox to the same one then we can save memory of 0x1000 bytes.

**Comment by RanderWang** [comment]
low priority
